### ISSUE TITLE
Enum weekdays

### ISSUE THREAD

**Issue by k3nsei** [FrozenNode/Laravel-Administrator#851]
I have enum field like:
```'week_day' => array(
            'title' => 'Dzień Tygodnia',
            'type' => 'enum',
            'options' => array(
                '1' => 'Poniedziałek',
                '2' => 'Wtorek',
                '3' => 'Środa',
                '4' => 'Czwartek',
                '5' => 'Piątek',
                '6' => 'Sobota',
                '7' => 'Niedziela'
            )
        ),```

But when i add or edit new item. Its saved value of the options not the key.

**Comment by camiloclc** [comment]
I'm having a similar issue, my enum keys are '1' and '2'. I've defined:

`
'forma_pagamento' => [
	'title' => 'Forma de pagamento',
	'type' => 'enum',
	'options' => [
		\App\Transacao::FORMA_PAGAMENTO_VISTA => 'Á vista',
		\App\Transacao::FORMA_PAGAMENTO_PARCELADO => 'Parcelado',
	],
],
`

in my model config for columns, edit_fields and filters.

In Fields/Enum.php, changing:

`
$options['options'][] = array(
	'id' => is_numeric($val) ? $text : $val,
	'text' => $text,
);
`

to:

`
$options['options'][] = array(
	'id' => (is_numeric($val) && is_string($val)) ? $text : $val,
	'text' => $text,
);
`

seems to resolve the issue.

By the way, the columns always show the enum's val, not the one I specified in the options array. Should I use an accessor or the output option or something like that? I'm having trouble finding this in the docs.
